pkg/controllers/nodeclaim/lifecycle/launch.go

```diff
@@ -61,7 +62,22 @@ func (l *Launch) Reconcile(ctx context.Context, nodeClaim *v1beta1.NodeClaim) (r
 		created, err = l.linkNodeClaim(ctx, nodeClaim)
 	} else {
 		created, err = l.launchNodeClaim(ctx, nodeClaim)
+		npKey := nodepoolutil.Key{Name: nodeClaim.Labels[v1beta1.NodePoolLabelKey], IsProvisioner: nodeClaim.IsMachine}
+		if npKey.Name != "" {
```

Comment:
```if npKey.Name != "" ``` this condition is added since tests panic . I think in real scenario it wun't be the case we can either leave as it is or apply nodepool in our tests wherever nodeclaim is applied